spark/src/main/java/org/apache/spark/CometTaskMemoryManager.java

```diff
@@ -48,6 +48,10 @@ public CometTaskMemoryManager(long id) {
   // Returns the actual amount of memory (in bytes) granted.
   public long acquireMemory(long size) {
     long acquired = internal.acquireExecutionMemory(size, nativeMemoryConsumer);
+    if (acquired < size) {
+      // If memory manager is not able to acquire the requested size, log memory usage
```

Comment:
perhaps it would be good to have some kind of a log with the reason?

Reply:
I'd like to go ahread and merge this PR so that we can start reviewing https://github.com/apache/datafusion-comet/pull/1730, which has many improvements and bug fixes.

Let's keep this conversation going, though, and handle it as a follow-up PR?